packages/iov-keycontrol/src/keyring.spec.ts

```diff
@@ -15,213 +15,274 @@ async function makeRandomEd25519Keypair(): Promise<Ed25519Keypair> {
 describe("Keyring", () => {
   const defaultChain = "chain123" as ChainId;
 
-  it("can be constructed", () => {
-    const keyring = new Keyring();
-    expect(keyring).toBeTruthy();
+  describe("constructor", () => {
+    it("can be constructed", () => {
+      const keyring = new Keyring();
+      expect(keyring).toBeTruthy();
+    });
+
+    it("is empty after construction", () => {
+      const keyring = new Keyring();
+      expect(keyring.getWallets().length).toEqual(0);
+    });
+
+    it("can deserialize empty", () => {
+      const keyring = new Keyring('{"formatVersion":1,"wallets":[]}' as KeyringSerializationString);
+      expect(keyring).toBeTruthy();
+      expect(keyring.getWallets().length).toEqual(0);
+    });
+
+    it("throws for unsupported format version", () => {
+      const data = '{"formatVersion":123,"wallets":[]}' as KeyringSerializationString;
+      expect(() => new Keyring(data)).toThrowError(/unsupported format version/i);
+    });
+
+    it("can deserialize one ed25519-hd wallet", () => {
```

Comment:
want to test deserializing two wallets with the same identity to cover the last place one could sneak in a duplicate?

Reply:
Thanks, done in 64c3230e440262c82f578fe060d433eb9fc4d1f0